<launch>
<!-- Launch file for Livox MID360 LiDAR -->

	<arg name="rviz" default="false" />
	<arg name="use_bag_time" default="0" />

	<rosparam command="load" file="$(find fast_lio_localization)/config/mid360.yaml" />

	<param name="feature_extract_enable" type="bool" value="0"/>
	<param name="point_filter_num" type="int" value="1"/>
	<param name="max_iteration" type="int" value="3" />
	<param name="filter_size_surf" type="double" value="0.5" />
	<param name="filter_size_map" type="double" value="0.5" />
	<param name="cube_side_length" type="double" value="1000" />
	<param name="runtime_pos_log_enable" type="bool" value="0" />
	<param name="pcd_save_enable" type="bool" value="0" />
	<param name="use_bag_time" type="bool" value="0" />

    <node pkg="fast_lio_localization" type="fastlio_mapping" name="laserMapping" output="screen">
		
	</node>

	<!-- <arg name="map" default="/home/unitree/teleoperation/nav/rosws/FAST_LIO_WS/src/FAST_LIO/PCD/scans.pcd" /> -->
	
	<!-- loalization-->
    <node pkg="fast_lio_localization" type="global_localization.py" name="global_localization" output="screen">
		<param name="use_bag_time" value="$(arg use_bag_time)" />
	</node>

	<!-- transform  fusion-->
    <node pkg="fast_lio_localization" type="transform_fusion.py" name="transform_fusion" output="screen">
		<param name="use_bag_time" value="$(arg use_bag_time)" />
	</node>

	<!-- 
	<include file="$(find fast_lio_localization)/launch/read_pcd_publish.launch">
		<arg name="pcd_file_path" value="/home/unitree/teleoperation/nav/rosws/FAST_LIO_WS/src/FAST_LIO/PCD/scans.pcd" />
        <arg name="publish_topic" value="/map3d" />
		<arg name="use_bag_time" value="$(arg use_bag_time)" />
  	</include> -->

	<!-- glbal map-->
    <!-- <node pkg="pcl_ros" type="pcd_to_pointcloud" name="map_publishe" output="screen"
		  args="$(arg map) 5 _frame_id:=map cloud_pcd:=/map3d" /> -->

	<!-- change map time-->
	<!-- <node pkg="fast_lio_localization" type="change_map_time.py" name="change_map_time" output="screen" 
		  param="frequency:=5"/> -->


	<group if="$(arg rviz)">
	<node launch-prefix="nice" pkg="rviz" type="rviz" name="rviz" args="-d $(find fast_lio_localization)/rviz_cfg/localization.rviz" />
	</group>

</launch>
